convert/source-ext2.c

```diff
@@ -489,7 +490,8 @@ static int ext2_create_symlink(struct btrfs_trans_handle *trans,
 	pathname = (char *)&(ext2_inode->i_block[0]);
 	BUG_ON(pathname[inode_size] != 0);
 	ret = btrfs_insert_inline_extent(trans, root, objectid, 0,
-					 pathname, inode_size + 1);
+					 pathname, inode_size + 1,
+					 0, inode_size + 1);
```

Comment:
`inode_size + 1` is here because I've added a separate `ram_bytes` param to `btrfs_insert_inline_extent`, because it's no longer necessarily the same as `size`.

Reply:
In fact this is a minor problem in the existing convert code. The resulted symlink would be one byte larger than the original path (but inode size is still correct).

E.g.
The original ext4 has a link to file name `path`:
```
lrwxrwxrwx 1 root root 4 Sep  3 10:27 path -> path
```

But after convert, the result looks like this:
```
	item 10 key (267 INODE_ITEM 0) itemoff 15543 itemsize 160
		generation 1 transid 1 size 4 nbytes 5
		block group 0 mode 120777 links 1 uid 0 gid 0 rdev 0
		sequence 0 flags 0x0(none)
		atime 1725325025.930020384 (2024-09-03 10:27:05)
		ctime 1725325023.110085974 (2024-09-03 10:27:03)
		mtime 1725325023.110085974 (2024-09-03 10:27:03)
		otime 1725325023.110085974 (2024-09-03 10:27:03)
	item 11 key (267 INODE_REF 256) itemoff 15529 itemsize 14
		index 3 namelen 4 name: path
	item 12 key (267 EXTENT_DATA 0) itemoff 15503 itemsize 26
		generation 4 type 0 (inline)
		inline extent data size 5 ram_bytes 5 compression 0 (none)
```
Note the link has one extra byte.

It's definitely not your fault nor it's causing problems, but it will also help a lot if you want to fix it.

Reply:
And the fix for that is here: https://github.com/kdave/btrfs-progs/pull/884

Reply:
The proper fix for that bug is here: https://github.com/kdave/btrfs-progs/pull/885

In fact the +1 is already causing `btrfs/012` random failures.
So that has to be fixed anyway.

Reply:
Okay, I'll rebase once you've committed your fixes.